x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts

```diff
@@ -564,6 +596,20 @@ export class KnowledgeBaseService {
     user?: { name: string; id?: string };
     namespace?: string;
   }): Promise<void> => {
+    // for now we want to limit the number of user instructions to 1 per user
+    if (document.type === KnowledgeBaseType.UserInstruction) {
+      const existingId = await this.getExistingUserInstructionId({
+        isPublic: document.public,
+        user,
+        namespace,
+      });
+
+      if (existingId) {
+        id = existingId;
+        document.doc_id = existingId;
+      }
```

Comment:
This is worrying. Why on earth do we have duplicate doc ids??

Reply:
IIRC, we have a keyword field of the document id to be able to search on it, and then we also have to deal with ids that are unique per space/user, but not per index. But I don't have all the details in my head, and any improvement here is more than welcome

Reply:
Imo the doc id should be a UUID. An entry can then have a human readable label/title that is searchable. 
I think the issues we have (duplicate id's, risk of overwriting other users entries) is because we conflated the two (unique id vs editable label)

Reply:
I'll into this in a follow-up next week. And yes, I'll make sure that the LLM can still update entries 🙂